integrations/watsonx/src/haystack_integrations/components/embedders/watsonx/document_embedder.py

```diff
@@ -0,0 +1,200 @@
+# SPDX-FileCopyrightText: 2025-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import Any
+
+from haystack import Document, component, default_from_dict, default_to_dict
+from haystack.utils import Secret, deserialize_secrets_inplace
+from ibm_watsonx_ai import Credentials
+from ibm_watsonx_ai.foundation_models import Embeddings
+
+
+@component
+class WatsonXDocumentEmbedder:
+    """
+    Computes document embeddings using IBM watsonx.ai models.
+
+    ### Usage example
+
+    ```python
+    from haystack import Document
+    from haystack_integrations.components.embedders.watsonx.document_embedder import WatsonXDocumentEmbedder
+
+    documents = [
+        Document(content='I love pizza!'),
+        Document(content='Pasta is great too'),
+    ]
+
+    document_embedder = WatsonXDocumentEmbedder(
+        model='ibm/slate-30m-english-rtrvr',
+        api_key=Secret.from_env_var('WATSONX_API_KEY'),
+        url='https://us-south.ml.cloud.ibm.com',
+        project_id='your-project-id',
+    )
+
+    result = document_embedder.run(documents=documents)
+    print(result['documents'][0].embedding)
+
+    # [0.017020374536514282, -0.023255806416273117, ...]
+    ```
+    """
+
+    def __init__(
+        self,
+        model: str = "ibm/slate-30m-english-rtrvr",
```

Comment:
Repeating this just in case it may be missed. 

For all component's let's enforce keyword arguments by making the following change
```suggestion
        self,
        *,
        model: str = "ibm/slate-30m-english-rtrvr",
```